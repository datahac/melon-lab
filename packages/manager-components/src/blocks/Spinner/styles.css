.spinner {
  position: relative;
}

.spinner__loader,
.spinner__loader::after {
  border-radius: 50%;
}

.spinner__loader {
  position: absolute;
  top: 0;
  border-style: solid;
  border-top-color: var(--color-black);
  border-right-color: var(--color-grey);
  border-bottom-color: var(--color-black);
  border-left-color: var(--color-grey);
  transform: translateZ(0);
  animation: load 2s infinite linear;
}

@-webkit-keyframes load {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

@keyframes load {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.spinner--default .spinner__loader,
.spinner--default :global(.spinner__icon) {
  width: 100px;
  height: 100px;
}

.spinner--default .spinner__loader {
  border-width: 2px;
}

.spinner--small .spinner__loader,
.spinner--small :global(.spinner__icon) {
  width: 50px;
  height: 50px;
}

.spinner--small .spinner__loader {
  border-width: 1px;
}

.spinner--inflated .spinner__loader,
.spinner--inflated :global(.spinner__icon) {
  width: 50px;
  height: 50px;
}

.spinner--inflated .spinner__loader {
  border-width: 4px;
}
