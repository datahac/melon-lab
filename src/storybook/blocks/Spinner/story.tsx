import { storiesOf } from '@storybook/react';
import React from 'react';
import Spinner from './index';

storiesOf('Blocks|Spinner', module)
  .add('Default', () => {
    return <Spinner icon />;
  })
  .add('Small', () => {
    return <Spinner icon size="small" />;
  })
  .add('without Icon', () => {
    return <Spinner size="inflated" />;
  });
