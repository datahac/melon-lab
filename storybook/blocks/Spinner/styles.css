.spinner {
  position: relative;
}

.spinner__loader,
.spinner__loader::after {
  border-radius: 50%;
}

.spinner__loader {
  position: absolute;
  top: 0;
  border-style: solid;
  border-top-color: var(--color-black);
  border-right-color: var(--color-grey);
  border-bottom-color: var(--color-black);
  border-left-color: var(--color-grey);
  transform: translateZ(0);
  animation: load 2s infinite linear;
}

.spinner__text {
  margin-top: var(--space-s);
  text-align: center;
}

@-webkit-keyframes load {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

@keyframes load {
  0% {
    transform: rotate(0deg);
  }
  100% {
    transform: rotate(360deg);
  }
}

.spinner__wrapper {
  margin: 0 auto;
}

.spinner--default .spinner__wrapper,
.spinner--default .spinner__loader,
.spinner--default :global(.spinner__icon) {
  width: 100px;
  height: 100px;
}

.spinner--default .spinner__loader {
  border-width: 2px;
}

.spinner--small .spinner__wrapper,
.spinner--small .spinner__loader,
.spinner--small :global(.spinner__icon) {
  width: 50px;
  height: 50px;
}

.spinner--small .spinner__loader {
  border-width: 1px;
}

.spinner--inflated .spinner__wrapper,
.spinner--inflated .spinner__loader,
.spinner--inflated :global(.spinner__icon) {
  width: 50px;
  height: 50px;
}

.spinner--inflated .spinner__loader {
  border-width: 4px;
}
