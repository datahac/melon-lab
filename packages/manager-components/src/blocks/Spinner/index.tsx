import React, { StatelessComponent } from 'react';
import Icon from '~/blocks/Icon';

import styles from './styles.css';

export interface SpinnerProps {
  size?: string;
  icon?: boolean;
}

const Spinner: StatelessComponent<SpinnerProps> = ({
  size = 'default',
  icon,
}) => {
  const sizeClass = `spinner--${size}`;
  const classes = `spinner ${sizeClass}`;

  return (
    <div className={classes}>
      <style jsx>{styles}</style>
      {icon &&
        <Icon iconClass="spinner__icon" name="logos_without-border" />
      }
      <div className="spinner__loader" />
    </div>
  );
};

export default Spinner;
