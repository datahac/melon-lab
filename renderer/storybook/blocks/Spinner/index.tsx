import classNames from 'classnames';
import React, { StatelessComponent } from 'react';
import Logos from '~/design/Logos';

import styles from './styles.css';

export interface SpinnerProps {
  icon?: boolean;
  size?: string;
  text?: string;
}

const Spinner: StatelessComponent<SpinnerProps> = ({
  icon,
  size = 'default',
  text,
}) => {
  const spinnerClassNames = classNames('spinner', {
    [`spinner--${size}`]: size,
  });

  return (
    <div className={spinnerClassNames}>
      <style jsx>{styles}</style>
      <div className="spinner__wrapper">
        {icon && <Logos logoClass="spinner__icon" name="without-border" />}
        <div className="spinner__loader" />
      </div>
      {text && <div className="spinner__text">{text}</div>}
    </div>
  );
};

export default Spinner;
