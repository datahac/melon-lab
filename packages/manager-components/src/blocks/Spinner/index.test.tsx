import React from 'react';
import Spinner from './index';

describe('Spinner', () => {
  const defaultElement = <Spinner />;

  it('should render correctly', () => {
    const wrapper = shallow(defaultElement);
    expect(wrapper).toMatchSnapshot();
  });
});
