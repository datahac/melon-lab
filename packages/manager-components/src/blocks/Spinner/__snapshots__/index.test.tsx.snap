// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner should render correctly 1`] = `
<div
  className="jsx-__scopedHash spinner spinner--default"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <div
    className="jsx-__scopedHash spinner__loader"
  />
</div>
`;
